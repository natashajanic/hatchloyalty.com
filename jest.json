{
  "collectCoverage": true,
  "coveragePathIgnorePatterns": ["tests"],
  "globals": {},
  "moduleFileExtensions": [
    "ts",
    "tsx",
    "js"
  ],
  "moduleNameMapper": {
    "^.+\\.(css|less|scss)$": "identity-obj-proxy",
    "\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/tests/assetsTransformer.ts",
    "^src/(.*)$": "<rootDir>/src/$1"
  },
  "setupFiles": [
    "raf/polyfill"
  ],
  "setupFilesAfterEnv": [
    "<rootDir>/tests/setupTests.ts"
  ],
  "snapshotSerializers": [
    "enzyme-to-json/serializer"
  ],
  "transform": {
    "\\.(ts|tsx)$": "ts-jest"
  },
  "testRegex": "/src/.*/__tests__/.*\\.(ts|tsx|js)$",
  "unmockedModulePathPatterns": [],
  "verbose": true
}
