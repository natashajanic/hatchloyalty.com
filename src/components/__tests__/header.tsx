import { shallow, ShallowWrapper } from 'enzyme'
import { Link } from 'gatsby'
import 'jest' // tslint:disable-line no-import-side-effect
import * as React from 'react'
import Headroom from 'react-headroom'
import Header, { IHeaderProps } from '../header'

describe('Header', () => {
  let props: IHeaderProps
  let mountedComponent: ShallowWrapper<any, any, Header>

  beforeEach(() => {
    props = {
      siteTitle: 'Hatch Loyalty',
      navLinks: [
        {
          name: 'Home',
          link: '/',
        },
        {
          name: 'Resources',
          link: '/resources',
        },
        {
          name: 'Blog',
          link: '/blogs',
        },
      ],
    }
    mountedComponent = shallow(<Header {...props} />)
  })

  describe('render', () => {
    it('renders all content in a sticky header wrapper', async () => {
      expect(mountedComponent.find(Headroom).length).toEqual(1)
    })

    it('renders the Hatch logo', async () => {
      expect(
        mountedComponent.find('Styled(Component)[is="img"]').length
      ).toEqual(1)
      const logo = mountedComponent.find('Styled(Component)[is="img"]').at(0)
      expect(logo.prop('alt')).toEqual(props.siteTitle)
    })

    it('renders each specified navigation link', async () => {
      expect(mountedComponent.find(Link).length).toEqual(
        1 + props.navLinks.length
      )
      props.navLinks.forEach((navLink, index) => {
        const link = mountedComponent.find(Link).at(index + 1)
        expect(link.text()).toEqual(navLink.name)
        expect(link.prop('to')).toEqual(navLink.link)
      })
    })

    it('matches the expected snapshot', async () => {
      expect(mountedComponent).toMatchSnapshot()
    })
  })
})
