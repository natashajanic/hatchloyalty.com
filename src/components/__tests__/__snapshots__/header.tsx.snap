// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header render matches the expected snapshot 1`] = `
<Headroom
  calcHeightOnResize={true}
  disable={false}
  disableInlineStyles={false}
  downTolerance={0}
  onPin={[Function]}
  onUnfix={[Function]}
  onUnpin={[Function]}
  parent={[Function]}
  pinStart={0}
  upTolerance={5}
  wrapperStyle={Object {}}
>
  <Styled(Component)
    alignItems="center"
    background="white"
    blacklist={
      Array [
        "css",
        "alignItems",
        "background",
        "border",
        "borderTop",
        "borderRight",
        "borderBottom",
        "borderLeft",
        "borderColor",
        "display",
        "justifyContent",
        "m",
        "mt",
        "mr",
        "mb",
        "ml",
        "mx",
        "my",
        "p",
        "pt",
        "pr",
        "pb",
        "pl",
        "px",
        "py",
        "width",
      ]
    }
    borderBottom="1px solid"
    borderColor="grayLight"
    display="flex"
    is="header"
    justifyContent="space-between"
    mx="auto"
    px={3}
    py={3}
    width="1024px"
  >
    <ForwardRef
      to="/"
    >
      <Styled(Component)
        alt="Hatch Loyalty"
        blacklist={
          Array [
            "css",
            "m",
            "mt",
            "mr",
            "mb",
            "ml",
            "mx",
            "my",
            "p",
            "pt",
            "pr",
            "pb",
            "pl",
            "px",
            "py",
            "width",
          ]
        }
        is="img"
        m={0}
        width="125px"
      />
    </ForwardRef>
    <Styled(Component)
      blacklist={
        Array [
          "css",
        ]
      }
      is="nav"
    >
      <Styled(Component)
        blacklist={
          Array [
            "css",
            "color",
            "bg",
            "fontSize",
            "fontWeight",
            "display",
            "m",
            "mt",
            "mr",
            "mb",
            "ml",
            "mx",
            "my",
            "p",
            "pt",
            "pr",
            "pb",
            "pl",
            "px",
            "py",
          ]
        }
        color="offBlack"
        display="inline-flex"
        fontSize={3}
        fontWeight={400}
        is="li"
        key="Home"
        px={2}
      >
        <ForwardRef
          style={
            Object {
              "color": "inherit",
              "textDecoration": "none",
            }
          }
          to="/"
        >
          Home
        </ForwardRef>
      </Styled(Component)>
      <Styled(Component)
        blacklist={
          Array [
            "css",
            "color",
            "bg",
            "fontSize",
            "fontWeight",
            "display",
            "m",
            "mt",
            "mr",
            "mb",
            "ml",
            "mx",
            "my",
            "p",
            "pt",
            "pr",
            "pb",
            "pl",
            "px",
            "py",
          ]
        }
        color="offBlack"
        display="inline-flex"
        fontSize={3}
        fontWeight={400}
        is="li"
        key="Resources"
        px={2}
      >
        <ForwardRef
          style={
            Object {
              "color": "inherit",
              "textDecoration": "none",
            }
          }
          to="/resources"
        >
          Resources
        </ForwardRef>
      </Styled(Component)>
      <Styled(Component)
        blacklist={
          Array [
            "css",
            "color",
            "bg",
            "fontSize",
            "fontWeight",
            "display",
            "m",
            "mt",
            "mr",
            "mb",
            "ml",
            "mx",
            "my",
            "p",
            "pt",
            "pr",
            "pb",
            "pl",
            "px",
            "py",
          ]
        }
        color="offBlack"
        display="inline-flex"
        fontSize={3}
        fontWeight={400}
        is="li"
        key="Blog"
        px={2}
      >
        <ForwardRef
          style={
            Object {
              "color": "inherit",
              "textDecoration": "none",
            }
          }
          to="/blogs"
        >
          Blog
        </ForwardRef>
      </Styled(Component)>
    </Styled(Component)>
  </Styled(Component)>
</Headroom>
`;
